assets/src/components/conversations/ConversationsProvider.tsx

```diff
@@ -417,47 +421,11 @@ export class ConversationsProvider extends React.Component<Props, State> {
       conversationsById: updatedConversationsById,
       messagesByConversation: updatedMessagesByConversation,
     };
-
-    // TODO: double check this logic
-    const conversationIds = Object.keys(updatedConversationsById).sort(
-      (x: string, y: string) => {
-        // TODO: DRY up this logic in other places
-        const a = updates.conversationsById[x];
-        const b = updates.conversationsById[y];
-        const left = a.last_activity_at
-          ? +new Date(a.last_activity_at)
-          : -Infinity;
-        const right = b.last_activity_at
-          ? +new Date(b.last_activity_at)
-          : -Infinity;
-
-        return right - left;
-      }
-    );
-
-    const allConversations = conversationIds.map(
-      (id) => updates.conversationsById[id]
-    );
-    const openConversations = allConversations.filter(
-      (c) => c.status === 'open'
-    );
-    const all = openConversations.map((c) => c.id);
-    const mine = openConversations
-      .filter((c) => c.assignee_id === currentUserId && c.status === 'open')
-      .map((c) => c.id);
-    const priority = openConversations
-      .filter((c) => c.priority === 'priority' && c.status === 'open')
-      .map((c) => c.id);
-    const closed = allConversations
-      .filter((c) => c.status !== 'open')
-      .map((c) => c.id);
+    const inboxes = this.getInboxes(updatedConversationsById);
```

Comment:
I've noticed that our codebase has a few pretty big functions here and there. I started to clean this up by creating a lot of smaller functions. Let me know what your thoughts are on many small functions vs fewer larger functions.

Reply:
yup i prefer this as well :+1: